src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -96,13 +157,386 @@ void AreaTrigger::Update(uint32 p_time)
         Remove(); // expired
 
     WorldObject::Update(p_time);
+    _timeSinceCreated += p_time;
+
+    if (GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_ATTACHED))
+    {
+        if (Unit* target = GetTarget())
+        {
+            GetMap()->AreaTriggerRelocation(this, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ(), target->GetOrientation());
+        }
+    }
+    else if (!_spline.empty())
+        UpdateSplinePosition();
+
+    sScriptMgr->OnAreaTriggerEntityUpdate(this, p_time);
+
+    switch (GetTemplate()->Type)
+    {
+        case AREATRIGGER_TYPE_SPHERE:
+            SearchUnitInSphere();
+            break;
+        case AREATRIGGER_TYPE_BOX:
+            SearchUnitInBox();
+            break;
+        case AREATRIGGER_TYPE_POLYGON:
+            SearchUnitInPolygon();
+            break;
+        case AREATRIGGER_TYPE_CYLINDER:
+            SearchUnitInCylinder();
+            break;
+        default:
+            break;
+    }
 }
 
 void AreaTrigger::Remove()
 {
     if (IsInWorld())
     {
+        if (Unit* caster = GetCaster())
+            caster->_UnregisterAreaTrigger(this);
+
+        for (ObjectGuid insideUnitGuid : _insideUnits)
+            if (Unit* leavingUnit = ObjectAccessor::GetUnit(*this, insideUnitGuid))
+                sScriptMgr->OnAreaTriggerEntityUnitExit(this, leavingUnit);
```

Comment:
that will not remove auras correctly, i think it is better to just do `HandleUnitEnterExit({});` here